main/app/logger/log.cpp

```diff
@@ -0,0 +1,77 @@
+#include "log.hpp"
+#include <time.hpp>
+
+#define LOG_FILE_FORMAT "%d_log.txt"
+#define MAX_FILE_BYTE_SIZE (1024 * 1000)
+
+extern std::string string_vformat(const char* apFormat, ...);
+
+static constexpr const char logIds[4] = {
+    'D',
+    'I',
+    'W',
+    'E'
+};
+
+static const char* resetStyle = "\x1b[0m";
+
+static const char* ansiColors[4] = {
+    "\x1b[38;5;120m",
+    "\x1b[38;5;250m",
+    "\x1b[38;5;220m",
+    "\x1b[38;5;196m"
+};
+
+bool currentFile = false;
+void CLog::m_RotateLogFile() {
+    m_Log = CFile(string_vformat(LOG_FILE_FORMAT, (currentFile ^= 1)));
+    m_Log.mWrite("");
+}
+
+CLog::CLog() : m_Log(string_vformat(LOG_FILE_FORMAT, currentFile)) {}
+
+CLog& CLog::getInstance() {
+    static CLog instance = {};
+    return instance;
+}
+
+void CLog::mInit() {}
```

Comment:
dit kan net zo goed in hpp?

Reply:
ja opzich wel ja, weet niet of we dat zo wel willen doen